modules/broker-amqp/src/main/java/io/ballerina/messaging/broker/amqp/AmqpConnectionManager.java

```diff
@@ -69,4 +70,21 @@ public void removeConnectionHandler(AmqpConnectionHandler handler) {
             return new ArrayList<>(connectionHandlers);
         }
     }
+
+    /**
+     * Closes an AMQP connection specified by the identifier.
+     *
+     * @param id the connection identifier
+     */
+    public void closeConnection(int id) {
+        synchronized (connectionHandlers) {
+            AmqpConnectionHandler connectionHandler
```

Comment:
Can't we use an ConcurrentHashMap for `connectionHandlers` and then use the `connectionHandlers.get(id)`?

All the other synchronization issues will be solved as well.